GPU/Software/RasterizerRectangle.cpp

```diff
@@ -76,7 +76,8 @@ template <bool alphaBlend>
 inline void DrawSinglePixel5551(u16 *pixel, const u32 color_in) {
 	u32 new_color;
 	// Because of this check, we only support src.a / 1-src.a blending.
-	if ((color_in >> 24) == 255 || !alphaBlend) {
+	// We take advantage of short circuiting by checking the constant (template) value first.
```

Comment:
Sometimes I have to trust compilers to do the most basic of optimizations, I will be very sad if compilers don't do at least this...

-[Unknown]

Reply:
Oh yeah, late night coding silliness. Since alphaBlend is a const here due to being a template argument, my thinking here is nonsense.